### PR TITLE
Support ability to find and update a ticket, including custom fields

### PR DESCRIPTION
**Pull request by richdrich**
Also add an integration test

### CODE REVIEW

File: lib/obeya/client.rb [hunk 1]
```diff
@@ -32,6 +38,20 @@ def create_ticket(title, description, format: 'text', ticket_type_name: 'Bug', b
...
 
+    def update_ticket(id, other_fields)
+      ticket = Ticket.new(other_fields)
+
+      response = put("/tickets/#{id}", ticket.to_json(custom_field_name_map, true))
+      case response.status
+        when 200..299
+          true
+        else
+          puts "status: #{response.status}"
```
Comment (alto) [inline lib/obeya/client.rb]: Please comment out those `puts` lines.

File: lib/obeya/client.rb [hunk 2]
```diff
@@ -66,8 +86,54 @@ def ticket_types
       end
     end
 
+    def tickets_in_named_bin(bin_name)
+      named_bin = bin(bin_name)
+      raise "Bin #{bin_name} not found" unless named_bin
+      tickets_in_bin(named_bin.id)
+    end
+
+    def tickets_in_bin(bin_id)
```
Comment (alto) [inline lib/obeya/client.rb:96]: Should that method be `private`?

File: lib/obeya/client.rb [hunk 3]
```diff
@@ -66,8 +86,54 @@ def ticket_types
...
+      end
+    end
+
+    def matching_tickets_in_named_bin(bin_name, title_matcher)
+      named_bin = bin(bin_name)
+      raise "Bin #{bin_name} not found" unless named_bin
+      matching_tickets_in_bin(named_bin.id, title_matcher)
+    end
+
+    def matching_tickets_in_bin(bin_id, title_matcher)
```
Comment (alto) [inline lib/obeya/client.rb:113]: `private`?

File: lib/obeya/client.rb [hunk 4]
```diff
@@ -66,8 +86,54 @@ def ticket_types
...
+      tickets = tickets_in_bin(bin_id)
+      case title_matcher
+      when String
+        tickets.select { |t| t.title == title_matcher }
+      when Regexp
+        tickets.select { |t| t.title =~ title_matcher }
+      end
+    end
+
+    # Get custom fields as an array of id => {id, name, type}
```
Comment (alto) [inline lib/obeya/client.rb]: But you return a `Hash`, don't you?

File: lib/obeya/client.rb [hunk 5]
```diff
@@ -1,27 +1,33 @@
 require 'faraday'
 require 'faraday_middleware'
+require 'date'
 
 module Obeya
   class Client
 
     OBEYA_ROOT_URL = "https://beta.getobeya.com"
+    CFTYPES = [::NilClass, ::String, ::Float, ::Integer, ::Array, ::Date]
```
Comment (alto) [inline lib/obeya/client.rb]: Please rename to `CUSTOM_FIELD_TYPES` to make clear what `CF` is.

File: lib/obeya/client.rb [hunk 6]
```diff
@@ -66,8 +86,54 @@ def ticket_types
...
+      case title_matcher
+      when String
+        tickets.select { |t| t.title == title_matcher }
+      when Regexp
+        tickets.select { |t| t.title =~ title_matcher }
+      end
+    end
+
+    # Get custom fields as an array of id => {id, name, type}
+    def custom_fields
```
Comment (alto) [inline lib/obeya/client.rb:124]: Should that be `private`?

File: lib/obeya/client.rb (hunk 2 above)
Comment (richdrich) [reply lib/obeya/client.rb:96]: I think it may be useful to a caller. Obeya bins are quite small.

Comment (alto) [reply lib/obeya/client.rb:96]: okay :)

File: lib/obeya/client.rb (hunk 6 above)
Comment (richdrich) [reply lib/obeya/client.rb:124]: I've exposed the API calls that I've implemented plus some utilities.

Comment (alto) [reply lib/obeya/client.rb:124]: Yeah, maybe you are right. I tend to have the smallest possible interface. But this is just a wrapper about what is provided by the API. So 👍

**Pull request merged**